### DISCUSSION TITLE
Welcome to TcUnit Discussions!

### DISCUSSION THREAD

**Discussion by sagatowski** [tcunit/TcUnit#127] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by YokMenns** [comment]
Hello Jacob,
in  the TcUnit version 1.1.0.0 you declared this method as PUBLIC. " FB_TestSuite.AreAllTestsFinished() "
After the implementation of these functions ( RUN_IN_SEQUENCE(FUN) and TEST_IN_ORDERED(FUN) in a branch, you now declared
this method as INTERNAL. Because i use this method outsite the TcUniit.
Can you declare this method as PUBLIC?
Thank you for the answer.
Have a nice day.
Yok Menns

**Comment by sagatowski** [comment]
Hi @YokMenns!

Then I will make sure to make the method public again.
Out of curiosity, can I ask you how you use this method?

**Comment by YokMenns** [comment]
Hi Jacob,
thank you.
I want to know when the testcase finished because i store some informations (like test time, test setup and so on) 
in an another xml file.

**Comment by Bjerg8k** [comment]
The Test_ORDERED is not implemented yet? When will you expect to release verison 1.2? And thanks for some great inspiration !

**Reply by sagatowski** [reply]
Hi @Bjerg8k! It is implemented but not yet fully tested/verified yet. You can download the latest source from main branch and compile it. I have also done some documentation and test projects:
https://tcunit.org/frequently-asked-questions/?Display_FAQ=970
https://github.com/tcunit/ExampleProjects/tree/master/RunTestsInSequenceExampleProjects

Could you test it and see if it works as expected?

**Comment by Bjerg8k** [comment]
That was a quick responds ! :=) I am new to github and I cant seem to figure out how to get the files?!

I downloaded the precompiled lib version 1.1 and it works without issues, other than the Test_Ordered is not there. I would like to use it test the new one out however.
Im working with servo/motion so my test needs to wait for one another.
Example:
TestPowerON
TestMoveAbsolute

And so forth 😊

Off topic:
Have you found a good way to use methods when we have stuff like ADSREAD or motion commands that can take several plc cycles to respond + only respond to a rising edge trig?

My initial solution is calling methods “MoveAbsolute”, this then call a statemachine on the function block that cyclic execute the needed blocks. This however means I don’t use the method itself for feedback.

BR
Jens Bjerg

**Reply by sagatowski** [reply]
@Bjerg8k No worries, we're all new at some point! Everything you need to know is documented here:
https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository

To create a library from source it's documented here:
https://infosys.beckhoff.com/english.php?content=../content/1033/tc3_plc_intro/4189255051.html&id=
